src/rust/engine/fs/store/src/remote/reapi.rs

```diff
@@ -0,0 +1,353 @@
+// Copyright 2022 Pants project contributors (see CONTRIBUTORS.md).
+// Licensed under the Apache License, Version 2.0 (see LICENSE).
+use std::cmp::min;
+use std::collections::{BTreeMap, HashSet};
+use std::convert::TryInto;
+use std::fmt;
+use std::sync::Arc;
+use std::time::{Duration, Instant};
+
+use async_oncecell::OnceCell;
+use async_trait::async_trait;
+use futures::{FutureExt, StreamExt};
+use grpc_util::retry::{retry_call, status_is_retryable};
+use grpc_util::{
+  headers_to_http_header_map, layered_service, status_ref_to_str, status_to_str, LayeredService,
+};
+use hashing::{Digest, Hasher};
+use protos::gen::build::bazel::remote::execution::v2 as remexec;
+use protos::gen::google::bytestream::byte_stream_client::ByteStreamClient;
+use remexec::{
+  capabilities_client::CapabilitiesClient,
+  content_addressable_storage_client::ContentAddressableStorageClient, BatchUpdateBlobsRequest,
+  ServerCapabilities,
+};
+use tokio::io::AsyncWriteExt;
+use tokio::sync::Mutex;
+use tonic::{Code, Request, Status};
+use workunit_store::{Metric, ObservationMetric};
+
+use super::{ByteSource, ByteStoreProvider, LoadDestination};
+
+#[derive(Clone)]
```

Comment:
Since this is now wrapped in a `Arc<dyn>`, it might not need to implement `Clone` anymore?

Reply:
Ah, yeah, I think you're right. I had to add it to minimise the diff of the migration (there was a few `self.clone()`) but can definitely be cleaned up now.